### PR TITLE
Create CODE_OF_CONDUCT.md

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
SUMMARY: None
Project governance thing, so it doesn't need to go in the changelog,

#### Purpose of change
Declaration of our lack of tolerance for intolerance in order to encourage participation from everyone.

Practically speaking, this does not change anything about how the project is governed, the standards and responsibilities stated here are in line with the standards and responsibilities that have been undertaken by the project maintainers since the inception of the project, this is merely a statement of intent to continue to uphold these principles.

#### Describe the solution
Announcing our stand on 
https://www.contributor-covenant.org/

##### Self-authored Code of Conduct or Set of Rules
Discarded because I don't consider myself to be a real expert in this area, and drafting these things seems to be rather tricky.
###### Unmodified version of the upstream Code of Conduct.
I made several alterations to the text of the document that I was uncomfortable with for various reasons.  The following lines were removed:
* Using welcoming and inclusive language
I agree with the *intent* of this clause, but as far as I can tell it is too poorly defined to be useful to add.
* Other conduct which could reasonably be considered inappropriate in a professional setting
This clause is also far too vague, plus the use of "professional" is problematic in that no one in the project is acting in a professional capacity.

And this line was edited:
* we as contributors and maintainers pledge to making participation in our project and our community a harassment-free experience for everyone
to
* we as maintainers pledge to making participation in our project and our community a harassment-free experience for everyone
I do not presume to make this pledge on behalf of anyone else.  This is a Code of Conduct I am comfortable enforcing, but I cannot state that everyone has or will make this pledge.
###### No Code of Conduct, or one that says, "what Kevin says goes".
This is roughly speaking the status quo.  If you've talked to me a significant amount you might have a reasonable idea of how I will act in a given situation.  It doesn't help at all if you haven't.

#### Additional context
This PR is intended to elicit feedback about the Code of Conduct.  Looking at the history of this kind of thing I fully expect it to be controversial.  Be civil, or your posts will be unceremoniously deleted.

### CODE REVIEW

**Comment by FulcrumA** [comment]
Aye, we had a big discussion about it just now on Discord. I maintain that the whole covenant is kinda questionable and given certain past projects experiences, should be approached with caution at the very least. I understand certain attitudes of wanting to prevent abuse, though and I am willing to maintain that simply letting moderation staff take care of cases of dickering causing disturbance and disrupting work on the project is enough on merely grounds of very widely-understood and practiced common sense and good manners and any official pledges of adherence to some particular political agenda or rules stemming from it may be unnecessarily constricting and abused. What constitutes harrasment vs disagreements is something moderators would have to decide no matter what the CoC says in every case of interpersonal problems, anyway.

The project is not political much in nature and allows people of various ideologies to work together at least as long as all participants strive to meet others halfway through and establish some sort of middle ground for constructive cooperation, without any single group demanding full submission to demands stemming from personal ideology. This works well and in fact may be one of the considerable contributing factors to current size of community and the speed of development and shouldn't be changed, but maintained as much as possible.

Or in other words, as imprecise and potentially loaded a statement as it may be: It's a game, dammit, let people enjoy and make it without pushing any highly polarized political agenda into it or having one feed on its entrails. We'll all be better for it.

**Comment by Robik81** [comment]
> Declaration of our lack of intolerance for intolerance

Nitpick, first intolerance should be tolerance?

Anyway, edits to original CoC seems to be quite reasonable.

**Comment by mlangsdorf** [comment]
I would prefer some variation of r/cataclysmdda's rules:
"1.  Don't be a dick."

I don't really trust the CoC organization, though the explicit words of the modified CoC are fine.  I don't have any knowledge of how many potential contributors we're turning away by not having an explicit CoC nor how many we'd add by having a simpler CoC versus this one.  But I feel shorter is better in this, since it's all going to be subjective and depend on Kevin's judgment anyway.

**Comment by nexusmrsep** [comment]
First: it's good that it's created.
But I also believe that while covenant's version is ok-ish when it comes to general and overall matters, it lacks a section for true governing rules of this project.

So while it's flowers and butterflies to say...

> Focusing on what is best for the community

... the true colors of governing here are:

-  setting direction of the development and project  standards are sole right of the project owner, so rules of democracy do not apply when it comes to defining direction of development, and setting standards that define the final or target shape of the project,
- ideas and projects that contradict standards set by project owner may be overruled at any time, even when endorsed by the substantial part of the community,
- project owner is the final judge and instance for direction of development, approving changes, game lore, inclusion and exclusion of mods and all things regarding what goes in or out of the project,
- project is open for all, but above rules and standards are strictly enforced for the sake of consistency and stability of the project, and it's set standards
- in situations of choice, when contradictory points of view or  different new paths of development collide or are being discussed and brainstormed, project owner has the final say and right to decide on those matters

Those are the rules I was expecting in the code of conduct, because they are the most often causes of misunderstandings or unnecessary emotions. There is a Latin saying: dura lex sed lex. Hard law, but law nevertheless. It must be expressed in my opinion or people will interpret this general code of conduct a in a way that each action against a standard-breaking idea or PR might be interpreted as maintainers breaking free and open access to the open project, and thus breaking their own code of conduct.

**Comment by kevingranade** [comment]
> Nitpick, first intolerance should be tolerance?

Whoops, fixed.

> I would prefer some variation of r/cataclysmdda's rules:
> "1. Don't be a dick."

I'm not super hostile to the concept, but instead I'd just leave the status quo of sufficiently bad behaviour gets warnings, potential post deletions, and in extreme cases bans.  Effectively it becomes, "don't do things that Kevin doesn't like", which *I'm* perfectly happy with, but it doesn't help potential contributors out much.

This outcome is not out of the question, though I haven't seen a particularly compelling argument about it.

> I don't really trust the CoC organization

I agree there seems to be some bad behaviour from the organization, which is what made me look carefully at the text and make a few edits instead of accepting it as written.  I'm perfectly happy to consider alternative texts, and I'm in NO rush to adopt one, so there's plenty of time to round up candidates if anyone is so inclined.\

> true governing rules of this project.

Governance and conduct are two different things.  I'm not against having something like docs/PROJECT_GOVERNANCE.md that roughly says what you did in your post, but it's not IMO related to this doc.

**Comment by Rivet-the-Zombie** [comment]
>I do not presume to make this pledge on behalf of anyone else. This is a Code of Conduct I am comfortable enforcing, but I cannot state that everyone has or will make this pledge.

Count me in - I'm entirely fine with this.

**Comment by OzoneH3** [comment]
It's kinda sad that such a document would even be needed in the first place.

```Declaration of our lack of tolerance for intolerance in order to encourage participation from everyone.```
So .. more people will contribute because of adding this?

In my oppinion this could still need some depoliticisation.

Why wouldn't this sentence just stop here:
```In the interest of fostering an open and welcoming environment, we as maintainers pledge to making participation in our project and our community a harassment-free experience for everyone.```
Isn't "everyone" already including all of the aftermentioned?

**Comment by Theundyingcode** [comment]
I agree with the points brought up by both @nexusmrsep and @mlangsdorf.

Unless there are specific reasons/precedent for having a specific rule, the rules governing conduct should be as short and simple as possible.  The cataclysm project is not a business that needs the legal protection provided by enforcing a CoC and simply setting out some basic expectations is enough.  IMHO, legalese will either put people off or implicitly allow something that is not acceptable but not specifically banned by the Coc. 

Ex:
1. Don't be a dick.
2. Arguing, debating and criticizing ideas is welcomed in moderation but, be supportive, criticizing or attacking people will not be tolerated.
3. No content that is racist, ... will not be accepted.
4. Developers (those with merge permission) have final say in determining violations of these rules and what the appropriate action/punishment should be.
5. Kevin shall be the final arbiter in disputes between developers as he is the God of the Cataclysm.

On the other hand providing a clear explanation of how the project is structured/governed like @nexusmrsep started laying out would be both a helpful reference for new contributors and provide understanding regarding the rejection of certain ideas.

**Comment by tyrael93** [comment]
No gods or kings
only kev

**Comment by woutershep** [comment]
"Practically speaking, this does not change anything about how the project is governed." IMO then don't do it. I forsee all its gonna do is give something to argue over whilst adding nothing.

You're gonna have to have infrastructure and people in place to handle issues anyways. People that are gonna break the new CoC need someone to warn them and to guide them to the right path. And then potentially to ban them when they continue to stray. If you have those kind of people, any CoC will be superflous and just be fuel for 'i know you said don't X but the CoC...'

Also if you do push ahead with this i think you need to be -very- clear wether this will only apply to interaction here on the repository or also game content.I think we should explicitly not police content the same as issues/prs & feedback on them. A lot of things we have / can do in game are probably in violation of it either literally or in spirit. Like burning down a refugee center with the refugees in it. The vibrator item breaks the no sexual things rule. The NPCs that threaten and cuss like sailors breaking the no rudeness rule. The 'red' house is basically one giant troll. Just to name a few, a explicit mention would go a long way to avoid such issues.

**Comment by Robik81** [comment]
> when people say "everyone" they actually mean "white dudes".

Never met white dude who would think that _everyone_ means just _white dudes_. 

Few stupid extremists exist, I am sure, but it seems to me that more stupid extremists actually want _everyone_ to mean _everyone but white dudes_.

**Comment by Rivet-the-Zombie** [comment]
>I seriously would suggest reconsidering making CoC changes on such grounds, as they're implying white users and moderators are considered by default racist and thus CoC has to balance their shortcomings.

I'm confused by this statement. Where does the code of conduct file imply that?

**Comment by FulcrumA** [comment]
@Rivet-the-Zombie 
I mean Kevin's explanation for particular change:
>if you want to actually send a message that you don't care about gender or race or nationality, you need to explicitly say so, because a lot of the time, when people say "everyone" they actually mean "white dudes".

Which may or not be a case in US but in general is pretty presumptious and rather unfair claim when made toward the community at large.

**Comment by OzoneH3** [comment]
```Not necessarily more people in raw numbers, but a greater variety of people.```
Is there somewhere some kind of evidence for this statement or anything to back this up?

```... because the status quo is that white dudes are normally the ones being targeted by everything ... when people say "everyone" they actually mean "white dudes"```
I don't even know how to react to this. I'm feeling very much offended by this and at the same ashamed that someone could even think that's true. Is this really what the world is becoming or is this actually only a US kinda thing?

```I'm confused by this statement. Where does the code of conduct file imply that?```
With the CoC in mind and what Kevin wrote, this is pretty much what it implies now.

**Comment by ZhilkinSerg** [comment]
Hey, I am a simple man: I see checklist - I wanna all checkboxes marked.

image

**Comment by Muffindrake** [comment]
US gender politics have no place in a software project, particularly not in one where 99% of the developers never even see each other in person. Leave that inside the campuses of the adult daycares that they stemmed from.

Given that the document that you're pointing at is already heavily-skewed towards that can of worms, I strongly suggest either picking nothing (my preferred choice) or literally anything that has not been written in the wake of entitedly-offended individuals standing up for people that haven't asked them to act in this way.

Any unwarranted racial (or other) slur would already be met with an instant deletion and ban from the project, so I have no idea what this change would even begin to address. There have been no contributions denied because a person was green and from Alpha Centauri, nor has harassment been an issue.

**Comment by kevingranade** [comment]
> CDDA community is people from all around the world not partaking in most of that insanity,

So you're saying part of the userbase DOES potentially "partake in that insanity" as the status quo.  Sounds like it's sometimes necessary, and harmless otherwise.

> in general is pretty presumptious and rather unfair claim when made toward the community at large.

I made no such claim.  I said that "saying 'everyone' but only really meaning your particular in-group *is a thing that happens*, therefore we want to make it clear that is not happening here".

>  Is there somewhere some kind of evidence for this statement or anything to back this up?

No, do I need that to publish a clarification that if someone is disparaging to other people or groups of people, there will be repercussions?

>  I'm feeling very much offended by this and at the same ashamed that someone could even think that's true.

I have no idea why you would be offended by the observation that pervasive racism and sexism exists, and it's centred around white guys, and sometimes it takes the form of silently excluding minorities instead of participating in overt racism.  Is any of this actually in question?

**Comment by Rivet-the-Zombie** [comment]
>Any unwarranted racial (or other) slur would already be met with an instant deletion and ban from the project, so I have no idea what this change would even begin to address.

If these proposed rules are just putting the current defacto project rules into official writing, then what's the issue here anyway?

**Comment by OzoneH3** [comment]
My only problem here is the sanctimonious (not sure if that's the right translation https://dict.leo.org/englisch-deutsch/scheinheilig) political correctness that this piece is based on. Otherwise I'm totally fine with the rules it's establishing.

I'll just leave this here and won't return to this PR: https://www.youtube.com/watch?v=tPEHbJgomgA

**Comment by dissociativity** [comment]
I'm for the CoC, the reactionary response purely based on whataboutism is diluting any sort of intellectual discussion of the CoC online, it's unfortunate and anything saying the CoC is fine gets downvoted to hell by reactionaries that are whataboutism-ing about the 'SJW menace'.

Even though it's TLDR; don't harrass other devs please, stop hitting on coders you find hot especially and generally don't be a dick.

**Comment by greenlucid** [comment]
I would be up for a completely customized Code of Conduct from scratch, done by the people who actually contribute to this specific project. I don't think it's fine for people outside the project to enforce the Code of Conduct. People who don't contribute due to a lack of CoC probably don't have much to contribute to the project, really. Signed, an user who hasn't contributed anything

**Comment by SomeDeadGuy** [comment]
>stop hitting on codders you find hot
Never! Fight me IRL!

**Comment by Kelenius** [comment]
I'll try to break down my thoughts on this. First, the current version - 

>Our Pledge

This part is perfectly reasonable, though I don't see why it needs to be written down.

**Comment by narc0tiq** [comment]
Don't mind me, I just need a Jenkins rebuild for testing. This comment will self-destruct in 10, 9...

**Comment by Kelenius** [comment]
Looking through the covenant, it seems to be either things covered by common sense or things too vague to have any point (e.g. "any contributor for other behaviors that they deem inappropriate, threatening, offensive, or harmful").

Furthermore, the creator of this covenant explicitly says that it's political: https://twitter.com/CoralineAda/status/1041465346656530432

It's a deep marsh that I would personally never deal with, and I don't think that getting into it is a good direction for this project. So I have to ask, @kevingranade, is this a direction you wish to take this project to? I think that a simpler and shorter covenant, not associated with politics, would be a better idea.

**Comment by Kelenius** [comment]
```We, the project maintainers, welcome diversity in all its forms, and we will not tolerate harassment or any other forms of inappropriate behavior. Such instances may be confidentially reported to kevin.granade@gmail.com.```

**Comment by ZhilkinSerg** [comment]
> We, the project maintainers, welcome diversity in all its forms, and we will not tolerate harassment or any other forms of inappropriate behavior. Such instances may be confidentially reported to kevin.granade@gmail.com.

Why would we welcome diversity? Okay, somebody is different, but how is it relevant to the project?

**Comment by Kelenius** [comment]
Yeah, that's probably reading too much into it. How about simply

```We, the project maintainers, do not tolerate harassment or any other forms of inappropriate behavior. Such instances may be confidentially reported to kevin.granade@gmail.com.```

**Comment by dissociativity** [comment]
'Welcome' doesn't mean 'encourage', it just means yeah, you're fine and safe here from assholes.
Plus, different perspectives lead to a larger variety of ideas, which would translate to more interesting PRs.

There's studies that say diversity does help projects, this isn't saying anywhere 'we'll literally accept your PR because you're a minority'.

It's just saying heyy, it's safe for you here and we won't tolerate people being assholes if your minority status comes up anywhere (it won't 99% of the time, but that's besides the point, having messages such as these make people more comfortable and not having that worry in the back of their mind that someone that knows of them /will/ bring that shit up and use it to encourage harassment, which is a real thing that can and does happen sometimes.

**Comment by ZhilkinSerg** [comment]
I don't know... Could be wrong, but I saw "encourage" connotation there. Why would someone bring up "minority" thing in the first place while it is totally unrelated to the project?

I do not see how it helps/hurts the project (doing a turn-based survival game set in a post-apocalyptic world) when contributors come up with their gender/background/food preference/political views (or anything else which makes people different.

**Comment by Voker57** [comment]
Can't see this CoC changing anything except making Kevin feel good about himself. All wording is maximally vague (except for very specific list of discrimination topics), and can only lead to people misinterpreting it and finding out only thing that still matters is what Kevin thinks.

**Comment by l29ah** [comment]
Don't forget to merge https://github.com/CleverRaven/Cataclysm-DDA/pull/23102 plz

**Comment by zcxv** [comment]
GNU recently publish [Kind Communications Guidelines](https://www.gnu.org/philosophy/kind-communication.html). This is one of the few guidelines that are well written imho.

**Comment by kevingranade** [comment]
> Kind Communication Guidelines

I am not opposed, even inclined to prefer them with some minor adjustments (removing the GNU specific political points).  Though I do have to point out that those guidelines seem to share every aspect people have objected to in this thread.

**Comment by dissociativity** [comment]
I don't see anything about forbidding and punishing?

**Comment by mlangsdorf** [comment]
I'd like to propose combining the pledge from the CC CoC with a summary of the GNU kind communications with links to a longer version:

#### Our Pledge
In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to making participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientation.

#### Our Standards
1. Assume other participants are posting in good faith, even if you disagree with what they say.
2. Make an effort to treat other participants with respect.
2a.  Do not take a harsh tone towards other participants, and especially don't make personal attacks against them.
2b. Recognize that criticism of your statements is not a personal attack on you.
2c. Avoid statements about the presumed typical desires, capabilities or actions of some demographic group.
3. Be especially kind to other contributors when saying they made a mistake.
4. Don't argue unceasingly for your preferred course of action when a decision for some other course has already been made.
5. If other participants complain about the way you express your ideas, please make an effort to cater to them. 
6. Don't raise unrelated political issues.
7. If you feel these standards are being violated, please alert Kevin Granade by (TBD, up to Kevin here).

That's about 15 lines of text, which about 5 lines too long, but I'm not sure what to cut other than the inclusion list in the Pledge - and I do feel that has some value.  But I'd still like to see a shorter CoC/guidelines so that people will actually read it.

**Comment by ZhilkinSerg** [comment]
> of age, body size, disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientation

Replace it with "of anything"?

**Comment by mlangsdorf** [comment]
The point of specifying everything in the inclusion list is to make sure that whatever a particular potential contributor's concern is, we are reassuring that we welcome people with that concern.

Saying "of anything" is short but vague.  Concerned people might still be concerned.

Maybe "we don't know or care if you're an especially intelligent canine or anything else as long as you contribute and meet the standards of behavior outlined below."

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+## Our Pledge
+
+In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to making participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientation.
+Our Standards
```
Comment (ZhilkinSerg) [inline CODE_OF_CONDUCT.md]:
```suggestion

## Our Standards
```

**Pull request merged**